### PR TITLE
Add tests to rules file

### PR DESCRIPTION
**Pull request by GeekyAubergine**
### Related Issues

Closes: #24 

### What's changed

- Added basic npm project set up
- Added jest to dev dependencies
- Written tests for `rules.js`
- Corrected potential divide by 0 error

### CODE REVIEW

**Comment by jacekkopecky** [comment]
Looks useful, thanks for starting this!

We will want to add :shirt: linting as well; and we may want to restructure the tests for rules – right now I can't focus on the tests properly but it looks a bit repetitive. `test.each()` might be useful, and the tests for `ruleA()` to `ruleC()` probably shouldn't mix with tests for the underlying stats functions.

**Comment by jacekkopecky** [comment]
thanks for those!

**Comment by GeekyAubergine** [comment]
> test.each() might be useful
I hadn't heard of that function before, it's much more readable, thank you

> the tests for ruleA() to ruleC() probably shouldn't mix with tests for the underlying stats functions
With the current structure, I don't think it is possible to get away from doing this. If `mean` could be extracted into another file the it could be appropriately mocked out to remove the mixing. I am unsure if that's something you would want to do.

> We will want to add 👕 linting as well
Would that not be more appropriate in another pull request as it will likely change all files making this pull request to become unreadable in the diff.

> why do you think that the mean of [] is 0?
I've never really considered it, I guess it's something I've always done. Would you prefer for `[]` to continue to throw a divide by zero exception or would you prefer it to throw an `invalid argument` exception?

**Comment by jacekkopecky** [comment]
Hi @GeekyAubergine, please keep the function `mean()` as simple as it can be, throwing divide by zero is OK as we never call it with an empty array. We don't need a test case for that.

Adding CI belongs nicely with adding tests, so if you want to add that workflow, please do. We can also do it ourselves. Typical on push in master, and on pull request.

Happy to add linting afterwards.

The rule tests can simply be moved to the end of the file. Anyway, can you move the test file out of `js/` and into `tests/`, please?

**Comment by GeekyAubergine** [comment]
I've reverted my changes to `mean()` and moved the test files

I've added github actions to run the CI as it seemed the most suitable candidate, you can see an example of the output here https://github.com/GeekyAubergine/dcalc/actions/runs/990867959 

As we can test against multiple versions of node do you want to raise the node version to 16 or leave it at 10?

File: js/rules.js [hunk 1]
```diff
@@ -111,7 +111,16 @@ function toClassification(mark) {
   return 'First-class honours';
 }
 
-
+module.exports = {
```
Comment (jacekkopecky) [inline js/rules.js:114]: will this raise an error in the browser?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+on: [push]
+
+jobs:
+  qa:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 14.x, 16.x]
```
Comment (jacekkopecky) [inline .github/workflows/ci.yml:11]: no need to waste resources, just test in the latest LTS version.

File: .nvmrc [hunk 3]
```diff
@@ -0,0 +1 @@
+10.16
```
Comment (jacekkopecky) [inline .nvmrc:1]: should be latest LTS version

File: README.md [hunk 4]
```diff
@@ -8,6 +8,11 @@ If you're a student, it may be interesting to see how your current/predicted gra
 
 Please add suggestions & bug reports to the [issue list](https://github.com/portsoc/dcalc/issues).
 
+## Running Tests
+- To run the tests you'll need to have node 10.16 installed, easiest way to get this is using [nvm](https://github.com/nvm-sh/nvm) if you're on Mac or Linux
```
Comment (jacekkopecky) [inline README.md:12]: should say "latest long-term support version" in some way, not 10.16

**Review by jacekkopecky — CHANGES_REQUESTED** [review]
thanks!

**Comment by GeekyAubergine** [comment]
Updated tests: https://github.com/GeekyAubergine/dcalc/runs/3580447987

**Comment by jacekkopecky** [comment]
Hi again, thanks for all the work, we're adding a bit more (if you don't do it, we eventually will, so no worries): can you please rebase this over the current state, and just use import/export for getting the functions you're testing?

**Pull request merged**